### PR TITLE
Create __add__ method for Moment

### PR DESCRIPTION
**Pull request by fedimser**
It will be an alias for with_operation method.

Fix for #2641 .

### CODE REVIEW

File: cirq/ops/moment.py [hunk 1]
```diff
@@ -193,6 +193,12 @@ def transform_qubits(self: TSelf_Moment,
     def _json_dict_(self):
         return protocols.obj_to_dict_helper(self, ['operations'])
 
+    def __add__(self, other):
+        if isinstance(other, raw_types.Operation):
+            return self.with_operation(other)
+        else:
```
Comment (vtomole) [inline cirq/ops/moment.py]: This `else` is unnecessary. Remove it.

File: cirq/ops/moment_test.py [hunk 2]
```diff
@@ -261,3 +261,11 @@ def test_immutable_moment():
         circuit = cirq.Circuit(cirq.X(q1))
         moment = circuit.moments[0]
         moment.operations += (cirq.Y(q2),)
+
+
+def test_add():
```
Comment (vtomole) [inline cirq/ops/moment_test.py]: Add a test that works for a non-list circuit  (e.g) `circuit3 = cirq.Circuit(cirq.CNOT(a, b), cirq.X(a))`

File: cirq/ops/moment.py (hunk 1 above)
Comment (fedimser) [reply cirq/ops/moment.py]: Done.

File: cirq/ops/moment_test.py (hunk 2 above)
Comment (fedimser) [reply cirq/ops/moment_test.py]: Done.

**Comment by fedimser** [comment]
I see. I put NotImplemented back, without `else`.
Actually, this is important, as failed test have shown (because `Moment` can also be added to `Circuit`, and it can be left operand).

**Comment by vtomole** [comment]
I can't update this branch to master. @fedimser Can you?

**Comment by fedimser** [comment]
Hi.
Can please someone with write aceess merge this pull request?

**Comment by vtomole** [comment]
@Strilanc Bot is down.

**Pull request merged**